diff -Nur qserialport-9999_orig/conf.pri qserialport-9999/conf.pri
--- qserialport-9999_orig/conf.pri	2010-10-26 02:26:35.000000000 +0000
+++ qserialport-9999/conf.pri	2010-10-26 02:27:04.000000000 +0000
@@ -1,7 +1,7 @@
 # qconf
 
-PREFIX = /usr/local
-BINDIR = /usr/local/bin
-DATADIR = /usr/local/share
+PREFIX = /usr
+BINDIR = /usr/bin
+DATADIR = /usr/share
 
 
diff -Nur qserialport-9999_orig/src/src.pro qserialport-9999/src/src.pro
--- qserialport-9999_orig/src/src.pro	2010-10-26 02:26:35.000000000 +0000
+++ qserialport-9999/src/src.pro	2010-10-26 02:28:52.000000000 +0000
@@ -114,9 +114,9 @@
 
 unix:!lib_bundle: {
   # install
-  QSERPORT_INSTALL_DIR = /usr/local/qserialport
+  QSERPORT_INSTALL_DIR = /usr
   QSERPORT_INSTALL_LIBDIR = $$QSERPORT_INSTALL_DIR/lib
   QSERPORT_INSTALL_INCDIR = $$QSERPORT_INSTALL_DIR/include
 
