diff -Nur dracut-026_orig/modules.d/98systemd/udevadm-cleanup-db.service dracut-026/modules.d/98systemd/udevadm-cleanup-db.service
--- dracut-026_orig/modules.d/98systemd/udevadm-cleanup-db.service	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026/modules.d/98systemd/udevadm-cleanup-db.service	2013-03-11 10:33:00.767341463 +0700
@@ -16,7 +16,7 @@
 
 [Service]
 Type=oneshot
-ExecStart=-/usr/bin/udevadm info --cleanup-db
+ExecStart=-/bin/udevadm info --cleanup-db
 StandardInput=null
 StandardOutput=null
 StandardError=null
