diff -Nur dracut-026_orig/modules.d/10i18n/console_init.sh dracut-026/modules.d/10i18n/console_init.sh
--- dracut-026_orig/modules.d/10i18n/console_init.sh	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026/modules.d/10i18n/console_init.sh	2013-03-11 10:12:17.630978044 +0700
@@ -4,8 +4,10 @@
 
 [ -n "$DRACUT_SYSTEMD" ] && exit 0
 
 if [ -x /lib/systemd/systemd-vconsole-setup ]; then
     /lib/systemd/systemd-vconsole-setup "$@"
+elif [ -x /usr/lib/systemd/systemd-vconsole-setup ]; then
+    /usr/lib/systemd/systemd-vconsole-setup "$@"
 fi
 
 [ -e /etc/vconsole.conf ] && . /etc/vconsole.conf
diff -Nur dracut-026_orig/modules.d/10i18n/module-setup.sh dracut-026/modules.d/10i18n/module-setup.sh
--- dracut-026_orig/modules.d/10i18n/module-setup.sh	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026/modules.d/10i18n/module-setup.sh	2013-03-11 10:12:03.854065819 +0700
@@ -12,7 +12,7 @@
 }
 
 install() {
-    [ -x /lib/systemd/systemd-vconsole-setup ] && dracut_install /lib/systemd/systemd-vconsole-setup
+    [ -x $systemdutildir/systemd-vconsole-setup ] && dracut_install $systemdutildir/systemd-vconsole-setup
     KBDSUBDIRS=consolefonts,consoletrans,keymaps,unimaps
     DEFAULT_FONT=LatArCyrHeb-16
     I18N_CONF="/etc/locale.conf"
diff -Nur dracut-026_orig/modules.d/90mdraid/module-setup.sh dracut-026/modules.d/90mdraid/module-setup.sh
--- dracut-026_orig/modules.d/90mdraid/module-setup.sh	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026/modules.d/90mdraid/module-setup.sh	2013-03-11 10:06:00.410405209 +0700
@@ -83,8 +83,8 @@
     inst_hook shutdown 30 "$moddir/md-shutdown.sh"
     inst_script "$moddir/mdraid-cleanup.sh" /sbin/mdraid-cleanup
     inst_script "$moddir/mdraid_start.sh" /sbin/mdraid_start
-    if [ -e /lib/systemd/system/mdmon@.service ]; then
-        inst_simple /lib/systemd/system/mdmon@.service
+    if [ -e $systemdsystemunitdir/mdmon@.service ]; then
+        inst_simple $systemdsystemunitdir/mdmon@.service
     fi
     inst_hook pre-shutdown 30 "$moddir/mdmon-pre-shutdown.sh"
 }
diff -Nur dracut-026_orig/modules.d/99base/init.sh dracut-026/modules.d/99base/init.sh
--- dracut-026_orig/modules.d/99base/init.sh	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026/modules.d/99base/init.sh	2013-03-11 10:10:57.466489690 +0700
@@ -37,8 +37,10 @@
 RD_DEBUG=""
 . /lib/dracut-lib.sh
 
 if [ -x /lib/systemd/systemd-timestamp ]; then
     RD_TIMESTAMP=$(/lib/systemd/systemd-timestamp)
+elif [ -x /usr/lib/systemd/systemd-timestamp ]; then
+    RD_TIMESTAMP=$(/usr/lib/systemd/systemd-timestamp)
 else
     read RD_TIMESTAMP _tmp < /proc/uptime
     unset _tmp
@@ -122,7 +122,8 @@
 source_hook pre-udev
 
 # start up udev and trigger cold plugs
-/lib/systemd/systemd-udevd --daemon --resolve-names=never
+if [ -x /lib/systemd/systemd-udevd ]; then /lib/systemd/systemd-udevd --daemon --resolve-names=never
+elif [ -x /usr/lib/systemd/systemd-udevd ]; then /usr/lib/systemd/systemd-udevd --daemon --resolve-names=never ; fi
 
 UDEV_LOG_PRIO_ARG=--log-priority
 UDEV_QUEUE_EMPTY="udevadm settle --timeout=0"
diff -Nur dracut-026_orig/modules.d/99base/module-setup.sh dracut-026/modules.d/99base/module-setup.sh
--- dracut-026_orig/modules.d/99base/module-setup.sh	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026/modules.d/99base/module-setup.sh	2013-03-11 10:11:10.442406721 +0700
@@ -49,7 +49,7 @@
     fi
 
     mkdir -p "${initdir}/var"
-    [ -x /lib/systemd/systemd-timestamp ] && inst /lib/systemd/systemd-timestamp
+    [ -x $systemdutildir/systemd-timestamp ] && inst $systemdutildir/systemd-timestamp
     if [[ $realinitpath ]]; then
         for i in $realinitpath; do
             echo "rd.distroinit=$i"
