diff -Nur dracut-026/modules.d/98systemd/module-setup.sh dracut-026_orig//modules.d/98systemd/module-setup.sh
--- dracut-026/modules.d/98systemd/module-setup.sh	2013-03-08 13:07:48.000000000 +0700
+++ dracut-026_orig//modules.d/98systemd/module-setup.sh	2013-03-11 16:03:29.137187641 +0700
@@ -202,6 +202,12 @@
         echo "RateLimitBurst=0"
     } >> "$initdir/etc/systemd/journald.conf"
 
+    # turn off systemd-modules-load.service in initrd
+    mkdir -p "$initdir/etc/systemd/system/systemd-modules-load.service.d/"
+    {
+        echo "[Unit]"
+        echo "ConditionPathExists=!/etc/initrd-release"
+    }  >> "$initdir/etc/systemd/system/systemd-modules-load.service.d/disable-in-initrd.conf"
 
 }
 
