Index: aksparpub.c
===================================================================
RCS file: /var/local/cvsroot/haspd/aksparlnx/aksparpub.c,v
retrieving revision 1.3
diff -u -p -u -p -u -r1.3 aksparpub.c
--- aksparlnx/aksparpub.c	3 May 2006 10:11:46 -0000	1.3
+++ aksparlnx/aksparpub.c	21 Feb 2007 20:07:43 -0000
@@ -76,9 +82,6 @@ unsigned int loglevel = 1;
 
 #if LINUX_VERSION_CODE >= 0x20600
 
-#include <linux/vermagic.h>
-MODULE_INFO(vermagic, VERMAGIC_STRING);
-
 static const char __module_depends[]
 __attribute_used__
 __attribute__((section(".modinfo"))) =
@@ -456,8 +473,7 @@ int init_module(void)
         }
 
 #if LINUX_VERSION_CODE >= 0x20600
-        if (loglevel >= 3)
-            printk(KERN_INFO "%s: found parport %u at %04lX\n", hardlock, i, (unsigned long)port->base);
+        printk(KERN_INFO "%s: found parport %u at %04lX\n", hardlock, i, (unsigned long)port->base);
 #endif
 
         hl_pports[hl_numports].port = port->base;
