diff -uNr ./aksparlnx-1.7-x86_64_orig/Makefile ./aksparlnx-1.7-x86_64/Makefile
--- ./aksparlnx-1.7-x86_64_orig/Makefile	2007-01-09 02:09:11.000000000 +0600
+++ ./aksparlnx-1.7-x86_64/Makefile	2009-12-15 11:07:23.000000000 +0600
@@ -38,9 +38,9 @@
 
 KERNSRC ?= /usr/src/linux-2.6.9
 
-ifdef TOPDIR
+#ifdef TOPDIR
 obj-m        = aksparlnx.o
-endif
+#endif
 aksparlnx-objs     =  aksparpub.o api.o
 
 # ------------------------------ 2.6.x ------------------------------
