diff --git a/aksparlnx/aksparpub.c b/aksparlnx/aksparpub.c
--- a/aksparlnx/aksparpub.c
+++ b/aksparlnx/aksparpub.c
@@ -58,9 +58,6 @@
 #if defined(__x86_64__) && LINUX_VERSION_CODE < 0x2060E
 #include <linux/ioctl32.h>
 #endif
-#if defined(__x86_64__) && LINUX_VERSION_CODE >= 0x2060E
-#include <linux/smp_lock.h>
-#endif
 #ifndef LINUX
 #define LINUX
 #endif
@@ -263,18 +260,14 @@
 
     switch(cmd) {
         case HLAPI_CALL:
-            lock_kernel();
             ret = hl_api_ioctl(file, cmd, arg, USER32);
-            unlock_kernel();
             break;
         default:
-            lock_kernel();
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 27)
             ret = hl_unlocked_ioctl(file, cmd, arg);
 #else
             ret = hl_ioctl(file->f_dentry->d_inode, file, cmd, arg);
 #endif
-            unlock_kernel();
             break;
     }
     return ret;
