From 05dcfcd69848e119c6a30d363bc41e896029f8af Mon Sep 17 00:00:00 2001
From: Sebastian Dröge <slomo@debian.org>
Date: Tue, 15 Dec 2009 12:24:25 +0000
Subject: Bug 588338 - Enable linux specific features on armel

---
diff --git a/configure.in b/configure.in
index b0bbaec..e4b684e 100644
--- a/configure.in
+++ b/configure.in
@@ -176,7 +176,7 @@ fi
 
 dnl check what OS we're on
 #AM_CONDITIONAL(HAVE_LINUX, test x$target_os = xlinux-gnu)
-if test x$target_os = xlinux-gnu; then
+if test x$target_os = xlinux-gnu -o x$target_os = xlinux-gnueabi; then
 	AC_DEFINE([HAVE_LINUX],[],[Whether we are using linux or not])
 fi
 
@@ -223,7 +223,7 @@ fi
 dnl check if inotify backend is enabled
 AM_CONDITIONAL(ENABLE_INOTIFY, test x$inotify = xtrue)
 
-if test x$os = xlinux-gnu; then
+if test x$os = xlinux-gnu -o x$os = xlinux-gnFrom 05dcfcd69848e119c6a30d363bc41e896029f8af Mon Sep 17 00:00:00 2001
From: Sebastian Dröge <slomo@debian.org>
Date: Tue, 15 Dec 2009 12:24:25 +0000
Subject: Bug 588338 - Enable linux specific features on armel

---
diff --git a/configure.in b/configure.in
index b0bbaec..e4b684e 100644
--- a/configure.in
+++ b/configure.in
@@ -176,7 +176,7 @@ fi
 
 dnl check what OS we're on
 #AM_CONDITIONAL(HAVE_LINUX, test x$target_os = xlinux-gnu)
-if test x$target_os = xlinux-gnu; then
+if test x$target_os = xlinux-gnu -o x$target_os = xlinux-gnueabi; then
 	AC_DEFINE([HAVE_LINUX],[],[Whether we are using linux or not])
 fi
 
@@ -223,7 +223,7 @@ fi
 dnl check if inotif