--- bmp/Makefile.am
+++ bmp/Makefile.am
@@ -1,4 +1,4 @@
-bmpdir = $(prefix)/share/SpaceAryarya/bmp
+bmpdir = $(BMP_PATH)
 bmp_DATA =   my.bmp enemy1.bmp shot.bmp bomb.bmp\
 	     enemy2.bmp kage.bmp wall.bmp wall2.bmp eshot.bmp\
 	     boss1_b.bmp  boss1_do.bmp  boss1_f.bmp\
--- configure.in
+++ configure.in
@@ -25,17 +25,20 @@
         BMP_PATH=/usr/local/share/SpaceAryarya/bmp
         WAV_PATH=/usr/local/share/SpaceAryarya/wav
 else
-        DATA_PATH=$prefix/share/SpaceAryarya/data
-        BMP_PATH=$prefix/share/SpaceAryarya/bmp
-        WAV_PATH=$prefix/share/SpaceAryarya/wav
+        DATA_PATH=$datadir/SpaceAryarya/data
+        BMP_PATH=$datadir/SpaceAryarya/bmp
+        WAV_PATH=$datadir/SpaceAryarya/wav
+        SCORE_PATH=$localstatedir/SpaceAryarya
 fi
 AC_DEFINE_UNQUOTED(DATA_PATH,"$DATA_PATH")
 AC_DEFINE_UNQUOTED(BMP_PATH,"$BMP_PATH")
 AC_DEFINE_UNQUOTED(WAV_PAT--- bmp/Makefile.am
+++ bmp/Makefile.am
@@ -1,4 +1,4 @@
-bmpdir = $(prefix)/share/SpaceAryarya/bmp
+bmpdir = $(BMP_PATH)
 bmp_DATA =   my.bmp enemy1.bmp shot.bmp bomb.bmp\
 	     enemy2.bmp kage.bmp wall.bmp wall2.bmp eshot.bmp\
 	     boss1_b.bmp  boss1_do.bmp  boss1_f.bmp\
--- configure.in
+++ configure.in
@@ -25,17 +25,20 @@
         BMP_PATH=/usr/local/share/SpaceAryarya/bmp
         WAV_PATH=/usr/local/share/SpaceAryarya/wav
 else
-        DATA_PATH=$prefix/share/SpaceAryarya/data
-        BMP_PATH=$prefix/share/SpaceAryarya/bmp
-        WAV_PATH=$prefix/share/SpaceAryarya/wav
+        DATA_PATH=$datadir/SpaceAryarya/data
+        BMP_PATH=$datadir/SpaceAryarya/bmp
+        WAV_PATH=$datadir/SpaceAryarya/wav
+        SCORE_PATH=$localstatedir/SpaceAryarya
 fi
 AC_DEFINE_UNQUOTED(DATA_PATH,"$DATA_PATH")
 AC_DEFINE_UNQUOTED(BMP_PATH,"$BMP_PATH")
 AC_DEFINE_UNQUOTED(WAV_PATH,"$WAV_PATH")
+AC_DEFINE_UNQUOTED(SCORE_PATH,"$SCORE_PATH")
 AC_DEFINE_UNQUOTED(TITLE,"$PACKAGE $VERSION")
 AC_SUBST(DATA_PATH)
 AC_SUBST(BMP_PATH)
 AC_SUBST(WAV_PATH)
+AC_SUBST(SCORE_PATH)
 
 AC_OUTPUT([data/Makefile wav/Makefile bmp/Makefile src/Makefile
 spacearyarya.spec Makefile])
--- data/Makefile.am
+++ data/Makefile.am
@@ -1,6 +1,7 @@
-datadir = $(prefix)/share/SpaceAryarya/data
+datadir = $(DATA_PATH)
 data_DATA  = stage1.dat stage2.dat
 EXTRA_DIST = stage1.dat stage2.dat .score
 
-install:
-	$(INSTALL) -c -m 666 ./.score $(DESTDIR)$(DATA_PATH)
+install-data-local:
+	$(mk