diff -ur mailer-0.7_orig/mailer.py mailer-0.7/mailer.py
--- mailer-0.7_orig/mailer.py	2011-02-01 09:47:18.000000000 +0000
+++ mailer-0.7/mailer.py	2013-12-02 05:32:51.000000000 +0000
@@ -30,7 +30,10 @@
 """
 import smtplib
 import threading
-import Queue
+try:
+    from Queue import Queue
+except ImportError: # Renamed in Python 3
+    from queue import Queue
 import uuid
 
 # this is to support name changes
