diff -ru raspberrypi-mkimage-0_p20120201.orig/imagetool-uncompressed.py raspberrypi-mkimage-0_p20120201/imagetool-uncompressed.py
--- raspberrypi-mkimage-0_p20120201.orig/imagetool-uncompressed.py	2013-07-19 15:11:23.373837900 +0200
+++ raspberrypi-mkimage-0_p20120201/imagetool-uncompressed.py	2013-07-19 15:31:56.495421758 +0200
@@ -3,15 +3,23 @@
 import os
 import re
 import sys
+import argparse
 
-try:
-   kernel_image = sys.argv[1]
-except:
-   kernel_image = ""
-
-if kernel_image == "":
-  print("usage : imagetool-uncompressed.py <kernel image>");
-  sys.exit(0)
+parser = argparse.ArgumentParser(description='Prepare kernel files for Raspberry Pi bootloader')
+parser.add_argument('--force', '-f', action='store_true', default=False,
+  help='overwrite target file')
+parser.add_argument('kernel',
+  help='kernel file from /usr/src/linux*/arch/arm/boot/Image')
+parser.add_argument('bootimage',
+  help='file to be placed in /boot/kdiff -ru raspberrypi-mkimage-0_p20120201.orig/imagetool-uncompressed.py raspberrypi-mkimage-0_p20120201/imagetool-uncompressed.py
--- raspberrypi-mkimage-0_p20120201.orig/imagetool-uncompressed.py	2013-07-19 15:11:23.373837900 +0200
+++ raspberrypi-mkimage-0_p20120201/imagetool-uncompressed.py	2013-07-19 15:31:56.495421758 +0200
@@ -3,15 +3,23 @@
 import os
 import re
 import sys
+import argparse
 
-try:
-   kernel_image = sys.argv[1]
-except:
-   kernel_image = ""
-
-if kernel_image == "":
-  print("usage : imagetool-uncompressed.py <kernel image>");
-  sys.exit(0)
+parser = argparse.ArgumentParser(descriptio