Index: krystaldrop/Sources/KDpp/Tools/FilePath.h
===================================================================
--- krystaldrop.orig/Sources/KDpp/Tools/FilePath.h
+++ krystaldrop/Sources/KDpp/Tools/FilePath.h
@@ -75,7 +75,7 @@ public:
 	/**
 		Adds a directory before the filePath....
 	*/
-	void KD_FilePath::PrefixDirectory(const string &path);
+	void PrefixDirectory(const string &path);
 
 
 	/**
Index: krystaldrop/Sources/KDpp/Video/Color.h
===================================================================
--- krystaldrop.orig/Sources/KDpp/Video/Color.h
+++ krystaldrop/Sources/KDpp/Video/Color.h
@@ -38,7 +38,7 @@ public:
 	KD_Color operator*( float lambda );
 
 	// Multiply by another color
-	KD_Color KD_Color::operator*( const KD_Color &other );
+	KD_Color operator*( const KD_Color &other );
 
 	// Sets all the Index: krystaldrop/Sources/KDpp/Tools/FilePath.h
===================================================================
--- krystaldrop.orig/Sources/KDpp/Tools/FilePath.h
+++ krystaldrop/Sources/KDpp/Tools/FilePath.h
@@ -75,7 +75,7 @@ public:
 	/**
 		Adds a directory before the filePath....
 	*/
-	void KD_FilePath::PrefixDirectory(const string &path);
+	void PrefixDirectory(const string &path);
 
 
 	/**
Index: krystaldrop/Sources/KDpp/Video/Color.h
===================================================================
--- krystaldrop.orig/Sources/KDpp/Video/Color.h
+++ krystaldrop/Sources/KDpp/Video/Color.h
@@ -38,7 +38,7 @@ public:
 	KD_Color operator*( float lambda );
 
 	// Multiply by another color
-	KD_Color KD_Color::operator*( const KD_Color &other );
+	KD_Color operator*( const KD_Color &other );
 
 	// Sets all the components between 0 and 1
 	void normalize();
Index: krystaldrop/Sources/KrystalDrop/Game/AnimRow.h
===================================================================
--- krystaldrop.orig/Sources/KrystalDrop/Game/AnimRow.h
+++ krystaldrop/Sources/KrystalDrop/Game/AnimRow.h
@@ -22,7 +22,7 @@ class KD_AnimatedRow: public KD_Row
    void UpdateBlocks (unsigned multiplier); /* multiplier is used to catch up when a lag has occured */
   
   public:
-   KD_AnimatedRow::KD_AnimatedRow (short Height_In_Gems, short x_Offset, 
+   KD_AnimatedRow (short Height_In_Gems, short x_Offset, 
                                    KD_Hand* Hand, KD_Parameters* Param, KD_Memo* Memo);
   ~KD_AnimatedRow();
   
Index: krystaldrop/Sources/KrystalDrop/Game/Table.h
===================================================================
--- krystaldrop.orig/Sources/KrystalDrop/Game/Table.h
+++ krystaldrop/Sources/KrystalDrop/Game/Table.h
@@ -329,7 +329,7 @@ public:
 	/**
 		Sets the sprites for the border of the field.
 	*/
-	v