diff -Nur systemd-198_orig/units/initrd-cleanup.service systemd-198/units/initrd-cleanup.service
--- systemd-198_orig/units/initrd-cleanup.service	2013-03-07 19:07:09.000000000 +0700
+++ systemd-198/units/initrd-cleanup.service	2013-03-11 15:04:26.253530304 +0700
@@ -10,8 +10,8 @@
 DefaultDependencies=no
 ConditionPathExists=/etc/initrd-release
 OnFailure=emergency.target
-Requires=local-fs.target swap.target
-After=local-fs.target swap.target
+Requires=initrd-fs.target swap.target
+After=initrd-fs.target swap.target
 
 [Service]
 Type=oneshot
diff -Nur systemd-198_orig/units/initrd-parse-etc.service systemd-198/units/initrd-parse-etc.service
--- systemd-198_orig/units/initrd-parse-etc.service	2013-03-07 19:07:09.000000000 +0700
+++ systemd-198/units/initrd-parse-etc.service	2013-03-11 15:08:44.677546796 +0700
@@ -8,13 +8,13 @@
 [Unit]
 Description=Reload Configuration from the Real Root
 DefaultDependencies=no
-Requires=local-fs.target swap.target
-After=local-fs.target swap.target
+Requires=initrd-fs.target swap.target sysroot.mount
+After=initrd-fs.target swap.target sysroot.mount
 OnFailure=emergency.target
 ConditionPathExists=/etc/initrd-release
 
 [Service]
 Type=oneshot
 ExecStartPre=/usr/bin/systemctl daemon-reload
-ExecStart=/usr/bin/systemctl start local-fs.target
+ExecStart=/usr/bin/systemctl start initrd-fs.target
 ExecStart=/usr/bin/systemctl --no-block start initrd-cleanup.service
diff -Nur systemd-198_orig/units/initrd-switch-root.target systemd-198/units/initrd-switch-root.target
--- systemd-198_orig/units/initrd-switch-root.target	2013-03-07 19:07:09.000000000 +0700
+++ systemd-198/units/initrd-switch-root.target	2013-03-11 15:04:26.267530305 +0700
@@ -12,5 +12,5 @@
 Requires=initrd-switch-root.service
 Before=initrd-switch-root.service
 AllowIsolate=yes
-Wants=initrd-udevadm-cleanup-db.service local-fs.target swap.target systemd-journald.service
-After=initrd-udevadm-cleanup-db.service local-fs.target swap.target emergency.service emergency.target
+Wants=initrd-udevadm-cleanup-db.service initrd-fs.target swap.target systemd-journald.service
+After=initrd-udevadm-cleanup-db.service initrd-fs.target swap.target emergency.service emergency.target
