From 39b83cdab37623a546344622db9bbbc784c15df5 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 04 Mar 2013 18:01:05 +0000
Subject: add initrd-fs.target and initrd-fs-pre.target

---
diff --git a/Makefile.am b/Makefile.am
index 5453868..e23a6b4 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -343,6 +343,8 @@ dist_systemunit_DATA = \
 	units/kexec.target \
 	units/local-fs.target \
 	units/local-fs-pre.target \
+	units/initrd-fs.target \
+	units/initrd-fs-pre.target \
 	units/remote-fs.target \
 	units/remote-fs-pre.target \
 	units/network.target \
diff --git a/src/core/special.h b/src/core/special.h
index 99c0e12..de5889d 100644
--- a/src/core/special.h
+++ b/src/core/special.h
@@ -48,6 +48,8 @@
 #define SPECIAL_SOCKETS_TARGET "sockets.target"
 #define SPECIAL_LOCAL_FS_TARGET "local-fs.target"
 #define SPECIAL_LOCAL_FS_PRE_TARGET "local-fs-pre.target"
+#define SPECIAL_INITRD_FS_TARGET "initrd-fs.target"
+#define SPECIAL_INITRD_FS_PRE_TARGET "initrd-fs-pre.target"
 #define SPECIAL_REMOTE_FS_TARGET "remote-fs.target"       /* LSB's $remote_fs */
 #define SPECIAL_REMOTE_FS_PRE_TARGET "remote-fs-pre.target"
 #define SPECIAL_SWAP_TARGET "swap.target"
diff --git a/units/initrd-fs-pre.target b/units/initrd-fs-pre.target
new file mode 100644
index 0000000..9aa8061
--- a/dev/null
+++ b/units/initrd-fs-pre.target
@@ -0,0 +1,11 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Initrd File Systems (Pre)
+Documentation=man:systemd.special(7)
+ConditionPathExists=/etc/initrd-release
diff --git a/units/initrd-fs.target b/units/initrd-fs.target
new file mode 100644
index 0000000..6ba1758
--- a/dev/null
+++ b/units/initrd-fs.target
@@ -0,0 +1,13 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Initrd File Systems
+Documentation=man:systemd.special(7)
+OnFailure=emergency.target
+OnFailureIsolate=yes
+ConditionPathExists=/etc/initrd-release
--
cgit v0.9.0.2-2-gbebe
