diff -Nur systemd-198_orig/units/initrd-udevadm-cleanup-db.service systemd-198/units/initrd-udevadm-cleanup-db.service
--- systemd-198_orig/units/initrd-udevadm-cleanup-db.service	2013-03-05 06:43:05.000000000 +0700
+++ systemd-198/units/initrd-udevadm-cleanup-db.service	2013-03-11 15:10:14.991545694 +0700
@@ -15,4 +15,4 @@
 
 [Service]
 Type=oneshot
-ExecStart=-/usr/bin/udevadm info --cleanup-db
+ExecStart=-/bin/udevadm info --cleanup-db
diff -Nur systemd-198_orig/units/systemd-udev-settle.service systemd-198/units/systemd-udev-settle.service
--- systemd-198_orig/units/systemd-udev-settle.service	2013-03-08 03:55:09.000000000 +0700
+++ systemd-198/units/systemd-udev-settle.service	2013-03-11 15:10:34.713544997 +0700
@@ -27,4 +27,4 @@
 Type=oneshot
 TimeoutSec=180
 RemainAfterExit=yes
-ExecStart=/usr/bin/udevadm settle
+ExecStart=/bin/udevadm settle
diff -Nur systemd-198_orig/units/systemd-udev-trigger.service systemd-198/units/systemd-udev-trigger.service
--- systemd-198_orig/units/systemd-udev-trigger.service	2013-03-08 03:55:09.000000000 +0700
+++ systemd-198/units/systemd-udev-trigger.service	2013-03-11 15:10:26.632545302 +0700
@@ -16,4 +16,4 @@
 [Service]
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=/usr/bin/udevadm trigger --type=subsystems --action=add ; /usr/bin/udevadm trigger --type=devices --action=add
+ExecStart=/bin/udevadm trigger --type=subsystems --action=add ; /bin/udevadm trigger --type=devices --action=add
